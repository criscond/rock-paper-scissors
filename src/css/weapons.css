.paper, .scissors, .rock, .lizzard, .spock {
  background-position: center center;
  background-repeat: no-repeat;
  background-size: contain;
  border-radius: 50%;
  height: 96px;
  margin:5px;
  width: 96px;
  transition: all .5s ease-out;
}

.weapons div:hover {
  box-shadow: 0px 0px 8px 1px var(--dark);
} 

.weapon h3 {
  font-size: 1rem;
  margin-top: 70%;
}

.paper {
  background-image: url('../images/paper.png');
}

.scissors {
  background-image: url('../images/scissors.png');
}

.rock {
  background-image: url('../images/rock.png');
}

.lizzard {
  background-image: url('../images/lizzard.png');
}

.spock {
  background-image: url('../images/spock.png');
}

.weapons {
	display: flex;
	flex-direction: column;
	flex-wrap: nowrap;
	justify-content: center;
	align-items: center;
}

.weapons figure img:hover{
	box-shadow: rgba(255, 255, 255, 0.2) 0 0 0 1px inset, rgba(0, 0, 0, 0.9) 0 0 0 1px;
}

/* .weapons-container {
	margin-top: 10%;
	text-align: center;
} */

@media (max-width: 639px) {
	.weapons figure img{
		padding: 0;
		width: 72px;
	}
}

@media (min-width: 640px) {
  .weapons {
    flex-direction: row;
  }

  .weapons h3 {
    font-size: 1.2rem;
    margin-top: 70%;
  }

  .paper, .scissors, .rock, .lizzard, .spock {
    cursor: pointer;
    height: 128px;
    margin: 10px;
    width: 128px;
  }  
}
