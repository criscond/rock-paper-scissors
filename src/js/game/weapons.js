import { gameData } from '../data';
import { getCurrentComplexity } from '.'

const getWeaponId = (weaponName) => {
  const { weapons } = gameData;

  return weapons.indexOf(weaponName);
}

const getWeaponName = (weaponId) => {
  const { weapons } = gameData;

  return weapons.slice(weaponId)[0];
}

const getMaxNoOfWeapons = () => {
  const { complexity } = gameData;
  const currentComplexity = getCurrentComplexity();

  return complexity[currentComplexity - 1].noOfWeapons;
}

const generateRandomWeaponId = () => Math.floor(Math.random() * getMaxNoOfWeapons());

export { generateRandomWeaponId, getMaxNoOfWeapons, getWeaponId, getWeaponName }
