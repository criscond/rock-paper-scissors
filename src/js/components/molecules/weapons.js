import { gameData } from '../../data';
import { createElement } from '../services';
import { getMaxNoOfWeapons } from '../../game';
import { makeTheChalenge } from '../../game';
import weapon from './weapon';

const weapons = () => {
  const { weapons } = gameData;
  const weaponsContainer = createElement({
    htmlTag: 'div',
    className: 'weapons',
    ariaLabel: 'Weapon selector',
    role: 'listbox'
  });

  const maxNoOfWeapons = getMaxNoOfWeapons();

  weapons.map((item, key) => {
    if (key < maxNoOfWeapons) {
      weaponsContainer.appendChild(weapon(item, makeTheChalenge));
    }
  });

  return weaponsContainer;
}

export default weapons;
