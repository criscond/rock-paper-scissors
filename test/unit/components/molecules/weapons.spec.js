import weapons from '../../../../src/js/components/molecules/weapons';
describe('Weapons', () => {

	it('should create a element', () => {
		expect(weapons()).toMatchSnapshot();
	});
});
