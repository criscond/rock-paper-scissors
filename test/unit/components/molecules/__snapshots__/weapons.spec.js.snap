// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weapons should create a element 1`] = `
<div
  class="weapons"
  role="listbox"
>
  <div
    class="scissors"
    role="option"
  >
    <h3
      class=""
    >
      Scissors
    </h3>
  </div>
  <div
    class="paper"
    role="option"
  >
    <h3
      class=""
    >
      Paper
    </h3>
  </div>
  <div
    class="rock"
    role="option"
  >
    <h3
      class=""
    >
      Rock
    </h3>
  </div>
</div>
`;
